File: src/main/java/com/sup/dev/java_pc/views/table/cells/ZTableCellCheck.java
package com.sup.dev.java_pc.views.table.cells;

import com.sup.dev.java.classes.callbacks.simple.Callback;
import com.sup.dev.java.libs.json.Json;
import com.sup.dev.java_pc.views.table.ZTableCell;
import com.sup.dev.java_pc.views.widgets.ZCheckBox;

import java.awt.*;

public class ZTableCellCheck extends ZTableCell {

    private final ZCheckBox checkBox;

    public ZTableCellCheck(int size, String label) {
        this(0, size, label);
    }

    public ZTableCellCheck(int type, int size, String label) {
        super(type, size, label);

        checkBox = new ZCheckBox(size, label);

        setValue(false, true);
    }

    @Override
    public void toJson(String key, Json json) {
        json.put(key, checkBox.isChecked());
    }

    @Override
    public void fromJson(String key, Json json) {
        checkBox.setChecked(json.getBoolean(key));
    }

    @Override
    public void setShowChanges() {

    }

    //
    //  Setters
    //


    @Override
    public void setEnabled(boolean b) {
        checkBox.setEnabled(b);
    }

    @Override
    public void setOnRightClick(Callback onRightClick) {

    }

    @Override
    protected void setValue(Object value) {
        checkBox.setChecked((Boolean)value);
    }

    //
    //  Getters
    //

    @Override
    public boolean isEmpty() {
        return checkBox.isChecked() == (Boolean) getOriginalValue();
    }

    @Override
    public void showIfError() {

    }

    @Override
    protected Object getCellValue() {
        return checkBox.isChecked();
    }

    @Override
    protected Component getView() {
        return checkBox;
    }

    @Override
    public boolean isError() {
        return false;
    }
}

File: src/main/java/com/sup/dev/java_pc/views/table/cells/ZTableCellCheck.kt
package com.sup.dev.java_pc.views.table.cells

import com.sup.dev.java.libs.json.Json
import com.sup.dev.java_pc.views.table.ZTableCell
import com.sup.dev.java_pc.views.widgets.ZCheckBox
import java.awt.Component


class ZTableCellCheck(type: Int, size: Int, label: String) : ZTableCell(type, size, label) {

    private val checkBox: ZCheckBox

    //
    //  Getters
    //

    override val isEmpty: Boolean
        get() = checkBox.isChecked == originalValue as Boolean?

    override val cellValue: Any
        get() = checkBox.isChecked

    override val view: Component
        get() = checkBox

    override val isError: Boolean
        get() = false

    constructor(size: Int, label: String) : this(0, size, label) {}

    init {

        checkBox = ZCheckBox(size, label)

        setValue(false, true)
    }

    override fun toJson(key: String, json: Json) {
        json.put(key, checkBox.isChecked)
    }

    override fun fromJson(key: String, json: Json) {
        checkBox.isChecked = json.getBoolean(key)
    }

    override fun setShowChanges() {

    }

    //
    //  Setters
    //


    override fun setEnabled(b: Boolean) {
        checkBox.isEnabled = b
    }

    override fun setOnRightClick(onRightClick: ()->Unit) {

    }

    override fun setValue(value: Any) {
        checkBox.isChecked = value as Boolean
    }

    override fun showIfError() {

    }
}
